File: rename_ui.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
    fs.readdirSync(dir).forEach(f => {
        let dirPath = path.join(dir, f);
        let isDirectory = fs.statSync(dirPath).isDirectory();
        if (isDirectory && !dirPath.includes('node_modules') && !dirPath.includes('.git')) {
            walkDir(dirPath, callback);
        } else if (!isDirectory) {
            callback(path.join(dir, f));
        }
    });
}

walkDir('d:\\Projects\\IS_SYSTEM_TEST', function(filePath) {
    if (filePath.match(/\.(html|js|css|ps1|sql)$/)) {
        let content = fs.readFileSync(filePath, 'utf-8');
        let matches = content.match(/.{0,30}%\s*[Бб][Рр][Аа][Кк].{0,30}/g);
        if (matches) {
            matches.forEach(m => console.log(`${path.basename(filePath)}: ${m}`));
        }
        let matches2 = content.match(/.{0,30}Процент Брак.{0,30}/gi);
        if (matches2) {
            matches2.forEach(m => console.log(`${path.basename(filePath)}: ${m}`));
        }
    }
});

File: rename_ui.py
import os
import re

files_to_check = []
for root, _, files in os.walk('d:\\Projects\\IS_SYSTEM_TEST'):
    if 'node_modules' in root or '.git' in root: continue
    for f in files:
        if f.endswith(('.html', '.js', '.css', '.ps1', '.sql')):
            files_to_check.append(os.path.join(root, f))

for filepath in files_to_check:
    try:
        with open(filepath, 'r', encoding='utf-8') as file:
            content = file.read()
            
        new_content = content
        
        # We will replace '% Брак' only in labels, headers, and UI messages.
        # Let's see all occurrences of % Брак first.
        matches = re.finditer(r'.{0,30}%\s*[Бб][Рр][Аа][Кк].{0,30}', content)
        for m in matches:
            print(f"{os.path.basename(filepath)}: {m.group(0)}")
            
    except Exception as e:
        pass
